### PR TITLE
add timeout for Hook

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

**Comment by vbatts** [comment]
LGTM

File: schema/defs.json [hunk 1]
```diff
@@ -114,6 +114,16 @@
...
+                },
+                "timeout": {
+                    "oneOf": [
+                        {
+                            "type": "integer"
+                        },
+                        {
+                            "type": "null"
+                        }
+                    ]
```
Comment (wking) [inline schema/defs.json]: We may want to add this `oneOf` as `intPointer` to match our other `*Pointer` definitions.

Comment (Mashimiao) [inline schema/defs.json]: fine, fixed.

**Comment by crosbymichael** [comment]
LGTM

**Comment by crosbymichael** [comment]
@vbatts you have to do it again because of the push

**Comment by wking** [comment]
Hmm, should timeout be unsigned?  I can't think of a use-case for
negative timeouts that isn't already covered by the field being
optional/nullable.

**Comment by vbatts** [comment]
@wking integer is fine for now.

**Comment by vbatts** [comment]
LGTM

**Pull request merged**